custom_components/tuya_local/devices/inkbird_ibt_26s_thermometer.yaml

```diff
@@ -0,0 +1,292 @@
+          - scale: 100
+        unit: F
+  - entity: sensor
+    name: Internal temperature
+    class: temperature
+    category: diagnostic
+    dps:
+      - id: 112
+        type: integer
+        name: sensor
+        unit: C
+        class: measurement
+
+  - entity: select
+    translation_key: temperature_unit
+    category: config
+    dps:
+      - id: 19
+        type: string
+        name: option
+        mapping:
+          - dps_val: F
+            value: fahrenheit
+          - dps_val: C
+            value: celsius
+  - entity: sensor
+    class: battery
+    category: diagnostic
+    dps:
+      - id: 101
+        type: integer
+        name: sensor
+        unit: "%"
+  - entity: number
+    name: Screen Brightness
+    category: config
+    icon: "mdi:brightness-percent"
+    dps:
+      - id: 102
+        type: integer
+        name: value
+        unit: '%'
+        range:
+          min: 0
+          max: 100
+  - entity: switch
+    name: Alarm Sound
+    category: config
+    dps:
+      - id: 104
+        type: boolean
+        name: switch
+  - entity: sensor
+    name: Status Code
+    category: diagnostic
+    dps:
+      - id: 128
+        type: integer
+        name: sensor
+
+  - entity: number
+    name: Calibration Probe 1
+    category: config
+    icon: "mdi:thermometer-check"
+    dps:
+      - id: 116
+        type: integer
+        name: value
+        unit: F
+        range:
+          min: -126
+          max: 126
+  - entity: number
+    name: Calibration Probe 2
+    category: config
+    icon: "mdi:thermometer-check"
+    dps:
+      - id: 117
+        type: integer
+        name: value
+        unit: F
+        range:
+          min: -126
+          max: 126
+  - entity: number
+    name: Calibration Probe 3
+    category: config
+    icon: "mdi:thermometer-check"
+    dps:
+      - id: 118
+        type: integer
+        name: value
+        unit: F
+        range:
+          min: -126
+          max: 126
+  - entity: number
+    name: Calibration Probe 4
+    category: config
+    icon: "mdi:thermometer-check"
+    dps:
+      - id: 119
+        type: integer
+        name: value
+        unit: F
+        range:
+          min: -126
+          max: 126
+  - entity: number
+    name: Calibration Probe 5
+    category: config
+    icon: "mdi:thermometer-check"
+    dps:
+      - id: 111
+        type: integer
+        name: value
+        unit: F
+        range:
+          min: -126
+          max: 126
+  - entity: number
+    name: Calibration Probe Oven
+    category: config
+    icon: "mdi:thermometer-check"
+    dps:
+      - id: 115
+        type: integer
+        name: value
+        unit: F
+        range:
+          min: -126
+          max: 126
+
+  - entity: select
+    name: Alarm Probe 1
+    category: diagnostic
+    icon: "mdi:alarm-light"
+    dps:
+      - id: 120
+        type: string
+        name: option
+        optional: true
+        mapping:
+          - dps_val: "EAAAAAAKBw=="
```

Comment:
I don't think the alarm is this simple. That is "On with some specific setting".  Maybe it is better to just leave the alarms as attributes on the temperature probe entities, and let the user set it in the official app, where all the functions are available.

Reply:
I don't use the official app, just Homeassistant, so I have the option of triggering the alarm sound for a sensor via automation. It works great, otherwise you would have to use multiple apps. For this reason it is optional.

Reply:
By contributing to this project, you are not just meeting your immediate needs, but making this available for everyone. So I don't agree with implementing this as "On" and "Off" with your personally preferred settings.  If you just want your own custom solution, you would be better using localtuya.

Reply:
I'm not merely "meeting my immediate needs;" rather, I have articulated a useful purpose for this type of function. The implementation doesn't necessarily have to be exactly as I described, but I haven't found any other way to trigger the device's alarm. I've now simply changed it to a switch. If it really shouldn't be there at all, I can remove it.

I'm open to suggestions on how to implement this or a similar function while also retaining the ability to trigger the alarm.

Reply:
Ideally I think if the bit that controls off/on can be determined, that can be exposed as an off / on switch.  But exposing it like now with a particular full setting of the alarm presumes that everyone will want the same settings as you, which I think is wrong.  I would rather it be a read-only attribute than to expose a switch that is geared to a single use case.

Reply:
The two settings you have now decode to the following 48bit values (HEX):

- ON  100000000007
- OFF 00aa05000007

Most likely that 10 in the first byte is the on/off bit, but it isn't clear why the following two bytes are different in the off value.

Reply:
Okay, I don't know enough about that, so I removed everything.